### PR TITLE
docs: add details about HPA via KEDA

### PR DESCRIPTION
**Pull request by thotz**
**Description of your changes:**
By default HPA can use details about memory or CPU consumption for
autoscaling, but also it can use custom metrics as well. There are alot
provides supports HPA via customer and one of them is KEDA project. Here
it is done with help of Prometheus Scaler

Signed-off-by: Jiffin Tony Thottan <thottanjiffin@gmail.com>
**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: Documentation/ceph-monitoring.md [hunk 1]
```diff
@@ -210,3 +210,30 @@ labels:
     prometheus: k8s
 [...]
 ```
+
+### Horizontal Pod Scaling using Kubernetes Event-driven Autoscaling aka KEDA
```
Comment (travisn) [inline Documentation/ceph-monitoring.md:214]:
```suggestion
### Horizontal Pod Scaling using Kubernetes Event-driven Autoscaling (KEDA)
```

File: Documentation/ceph-monitoring.md [hunk 2]
```diff
@@ -210,3 +210,30 @@ labels:
     prometheus: k8s
 [...]
 ```
```
Comment (travisn) [inline Documentation/ceph-monitoring.md:212]: It would be nice to have a table of contents at the top so new topics like this can be more easily discovered, but that is really a separate question from this PR.

File: Documentation/ceph-monitoring.md [hunk 3]
```diff
@@ -210,3 +210,30 @@ labels:
...
+
+### Horizontal Pod Scaling using Kubernetes Event-driven Autoscaling aka KEDA
+
+Using metrics exported from the Prometheus service, the horizontal pod scaling can use the custom metrics other than CPU and memory consumption. It can be done with help of Promethues Scaler provided by the [KEDA](https://keda.sh/docs/1.4/scalers/prometheus/). The KEDA deployed via helm chart, refer [here](https://keda.sh/docs/2.4/deploy/) for details.
```
Comment (travisn) [inline Documentation/ceph-monitoring.md:216]: These version links will get outdated as they have more releases. Is there a "latest" or similar link we can use?
```suggestion
Using metrics exported from the Prometheus service, the horizontal pod scaling can use the custom metrics other than CPU and memory consumption. It can be done with help of Prometheus Scaler provided by [KEDA](https://keda.sh/docs/2.4/scalers/prometheus/). See the [KEDA deployment guide](https://keda.sh/docs/2.4/deploy/) for details.
```

File: Documentation/ceph-monitoring.md [hunk 4]
```diff
@@ -210,3 +210,30 @@ labels:
...
+
+Following is sample for `ScaledObject`for autoscaling RGW:
```
Comment (travisn) [inline Documentation/ceph-monitoring.md:218]:
```suggestion
Following is an example to autoscale RGW:
```

File: Documentation/ceph-monitoring.md [hunk 5]
```diff
@@ -210,3 +210,30 @@ labels:
...
+```YAML
```
Comment (travisn) [inline Documentation/ceph-monitoring.md:219]: Please also create this yaml in the [monitoring examples folder](https://github.com/rook/rook/tree/master/cluster/examples/kubernetes/ceph/monitoring).

**Review by travisn — CHANGES_REQUESTED** [review]

**Review by travisn — COMMENTED** [review]
The doc looks good, the yaml linter just has a few whitespace issues to fix.

**Comment by thotz** [comment]
> The doc looks good, the yaml linter just has a few whitespace issues to fix.

May I skip validate-yaml for the keda.yaml file :
```diff
diff --git a/tests/scripts/github-action-helper.sh b/tests/scripts/github-action-helper.sh
index 25b717af8..aea4f1325 100755
--- a/tests/scripts/github-action-helper.sh
+++ b/tests/scripts/github-action-helper.sh
@@ -132,8 +132,8 @@ function build_rook_all() {
 function validate_yaml() {
   cd cluster/examples/kubernetes/ceph
   kubectl create -f crds.yaml -f common.yaml
-  # skipping folders and some yamls that are only for openshift.
-  manifests="$(find . -maxdepth 1 -type f -name '*.yaml' -and -not -name '*openshift*' -and -not -name 'scc*')"
+  # skipping folders and some yamls that are only for openshift, keda etc.
+  manifests="$(find . -maxdepth 1 -type f -name '*.yaml' -and -not -name '*openshift*' -and -not -name 'scc*' -and -not -name 'keda*')"
   with_f_arg="$(echo "$manifests" | awk '{printf " -f %s",$1}')" # don't add newline
```
Otherwise I keda need to be installed by deafult

**Review by travisn — CHANGES_REQUESTED** [review]
@thotz Instead of skipping the yaml validation, how about following the pattern we use for the volume replication CRDs and install them as part of the [validate_yaml()](https://github.com/rook/rook/blob/b3b8172e13d967022ea62608bd413236c23aa1ed/tests/scripts/github-action-helper.sh#L164) test?

**Review by travisn — APPROVED** [review]

File: tests/scripts/github-action-helper.sh [hunk 6]
```diff
@@ -173,6 +173,11 @@ function validate_yaml() {
   kubectl create -f "${replication_url}/replication.storage.openshift.io_volumereplications.yaml"
   kubectl create -f "${replication_url}/replication.storage.openshift.io_volumereplicationclasses.yaml"
 
+  #create the KEDA CRDS
+  keda_version=2.4.0
+  keda_url="https://github.com/kedacore/keda/releases/download/v${keda_version}/keda-${keda_version}.yaml"
+  kubectl apply -f "${keda_url}"
+
```
Comment (BlaineEXE) [inline tests/scripts/github-action-helper.sh:180]: @thotz Why was this included but no integration test written to test KEDA integration with Rook RGWs?

**Review by BlaineEXE — COMMENTED** [review]
@thotz did you manually test that Rook RGWs support automatic scale-out like this?

I am concerned that this won't really work with Rook, because any time the CephObjectStore is reconciled, Rook will reset the replicas to 1.
